patches/playwright-core+1.24.0.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/node_modules/playwright-core/lib/server/fetch.js b/node_modules/playwright-core/lib/server/fetch.js
+index 13cd767..f9d42d2 100644
+--- a/node_modules/playwright-core/lib/server/fetch.js
++++ b/node_modules/playwright-core/lib/server/fetch.js
+@@ -187,7 +187,9 @@ class APIRequestContext extends _instrumentation.SdkObject {
+       const cookie = parseCookie(header);
+       if (!cookie) continue; // https://datatracker.ietf.org/doc/html/rfc6265#section-5.2.3
+ 
+-      if (!cookie.domain) cookie.domain = url.hostname;else (0, _utils.assert)(cookie.domain.startsWith('.'));
++      // This hack allows setting cookies in site/tests/shared/nav.ts → login
++      // To be removed in https://app.asana.com/0/1202542409311090/1202652337622563/f
++      if (!cookie.domain) cookie.domain = url.hostname;else cookie.domain = cookie.domain.replace(/^./, '');
```

Comment:
I faced a rather odd Playwright bug that was not allowing me to set cookies. This patched helped avoid this issue. TODO (later): investigate this code and submit upstream issue if needed: https://github.com/microsoft/playwright/blame/99d1ad5a88c3e89360829eee92dbaa98d75beaa4/packages/playwright-core/src/server/fetch.ts#L210-L213

Tracked in https://app.asana.com/0/1202542409311090/1202652337622563/f (internal)